uportal-war/src/main/java/org/jasig/portal/permission/target/EntityTargetProviderImpl.java

```diff
@@ -97,18 +142,14 @@ public IPermissionTarget getTarget(String key) {
          * possible that this implementation would demonstrate inconsistent
          * behavior.
          */
-        
-        for (String type : allowedEntityTypes) {
-            JsonEntityBean entity = groupListHelper.getEntity(type, key, false);
+        for (TargetType targetType : allowedTargetTypes) {
+            JsonEntityBean entity = groupListHelper.getEntity(targetType.toString(), key, false);
             if (entity != null) {
-                String entityId = entity.getId();
-                if("portlet".equals(type))
-                    entityId = IPermission.PORTLET_PREFIX.concat(entityId);
-                IPermissionTarget target = new PermissionTargetImpl(entityId, entity.getName());
+                IPermissionTarget target = new PermissionTargetImpl(entity.getId(), entity.getName(), targetType);
                 return target;
```

Comment:
That's right -- in all cases that I observed, the PORTLET_PREFIX had allready been applied to the entityId, and so this code was duplicating it.